ledger/ledger-consensual/src/main/kotlin/net/corda/v5/ledger/consensual/Party.kt

```diff
@@ -1,29 +1,35 @@
 package net.corda.v5.ledger.consensual
 
 import net.corda.v5.base.annotations.CordaSerializable
-import net.corda.v5.base.annotations.DoNotImplement
 import net.corda.v5.base.types.MemberX500Name
 import net.corda.v5.crypto.toStringShort
 import java.security.PublicKey
 
 /**
  * Representation of a Party.
  */
-@DoNotImplement
 @CordaSerializable
-interface Party {
+class Party(
     /**
      * @property name X500 Name of the party.
      */
-    val name: MemberX500Name
-
+    val name: MemberX500Name,
     /**
      * @property owningKey The [PublicKey] owned by the party.
      */
     val owningKey: PublicKey
+) {
 
     /**
      * Concise, readable String representation of a Party.
      */
     val description: String get() = "$name (owningKey = ${owningKey.toStringShort()})"
+
+    override fun toString() = name.toString()
+    override fun equals(other: Any?): Boolean =
+        other === this ||
+                other is Party &&
+                other.owningKey == owningKey &&
+                other.name == name
+    override fun hashCode(): Int = owningKey.hashCode()
```

Comment:
@mnesbit Would there ever be a case where `Party` might be constructed with different `MemberX500Name`s, but the same `PublicKey`?

Just wondering if `hashCode` should account for the `name` property as well; for example:

```kotlin
override fun hashCode(): Int = Objects.hash(name, owningKey)
```

Reply:
I think I got that part from here:
https://github.com/corda/corda-runtime-os/blob/release/os/5.0/libs/membership/ledger-identity/src/main/kotlin/net/corda/membership/ledger/identity/PartyImpl.kt#L25

Reply:
(These changes got reverted here.)